doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,201 @@
+.. _net_app_api:
+
+Network Application API
+#######################
+
+The Network Application API allows applications to
+
+* **Initialize** the application for networking use. This means for example
+  that if the application needs to have an IPv4 address, and if DHCPv4 is
+  enabled, then the net-app API will make sure that the device will get an
+  IPv4 address before the application is started.
+
+* **Set** various options for the networking subsystem. This means that if the
+  user has set options like IP addresses, IEEE 802.15.4 channel etc. in the
+  project configuration file, then those settings are applied to the system
+  before application starts.
+
+* **Create** a simple TCP/UDP server or client application. The net-app API
+  has functions that make it easy to create a simple TCP or UDP based network
+  application. The net-app API also provides transparent TLS and DTLS support
+  for the application.
+
+The net-app API functionality is enabled by `CONFIG_NET_APP` option.
+The current net-app API implementation is still experimental and may change and
+improve in future releases.
+
+Initialization
+##############
+
+The net-app API provides `net_app_init()` function that can configure the
+networking subsystem for the application. Following configuration options
+control how the configuration is done:
+
+* **CONFIG_NET_APP_AUTO_INIT** automatically configures the system according
```

Comment:
you can link to the Kconfig option documentation with:

     :option:`CONFIG_NET_APP_AUTO_INIT``